### DISCUSSION TITLE
posts/caching-tokens-supabase

### DISCUSSION THREAD

**Discussion by giscus** [laymonage/isle#40] [Comments]
# posts/caching-tokens-supabase

To help prevent giscus from hitting GitHub API's rate limit, we cache the access tokens in Supabase. Here's how I did it.

https://laymonage.com/posts/caching-tokens-supabase

**Comment by umschaudhary** [comment]
Awesome!

**Comment by lattechen925** [comment]
1

**Comment by matschik** [comment]
Interesting !

**Comment by unphased** [comment]
Thanks for explaining how this works! I am eagerly looking forward to implementing Giscus when I resurrect my blog site, and I have a few simple questions. 

1. The use of Supabase: is it something you have to implement one time in Giscus and the tokens cached by your free Supabase instance will support all Giscus users? 500MB may be limiting? Or maybe only the very latest few tokens need to be stored for everyone. Otherwise, It sounds like we all need to implement our own database caching for when we host giscus. 

2. I skimmed around very quickly on your repo and it looks like your content repo holding your actual blog's content is a submodule of your base repo, and the base repo is what holds github discussions that map to giscus, this is all very awesome but I am curious why some of the embedded images resolve to 404's when browsing your repo in github while they appear fine here on your actual blog page. 

I have been planning to host static content for my sites out of cloudflare backed by S3, but I definitely think tracking blog content in git is the way to go so I was planning to script stuff to clone git repo content to s3 but maybe it is better to leave s3 out of the equation. How do you go about converting your content repo into your website? Just looking for a very rough overview :)

Thanks!!!

**Comment by yxsky88** [comment]
111
